backend/src/test/java/de/armbrust/planz/service/ProductServiceTest.java

```diff
@@ -124,12 +126,105 @@ public void testAddingListOfNotExistingProducts() throws CryptoException, Missin
         when(amazonApiHead.getProductsFromApiReport()).thenReturn(products);
 
         //WHEN
-        productService.UpdateProductDb();
+        productService.initializeProductsOnDb();
 
         //THEN
         verify(productMongoDb).save(product1);
         verify(productMongoDb).save(product2);
     }
 
+    @Test
+    @DisplayName("if product available the inventory should be add to inventory array")
+    public void IfProductAvailibeInventoryShouldAdded() {
+        //GIVEN
+        Product product1 = Product.builder()
+                .sku("123")
+                .title("Product1")
+                .asin("ABC")
+                .inventory(new ArrayList<>())
+                .build();
+
+        Inventory inventory1 = Inventory.builder()
+                .warehouse("AnyWarehouse")
+                .condition("NEW")
+                .country("DE")
+                .amount("23")
+                .date("2021-02-25")
+                .sku("123")
+                .build();
+
+        when(productMongoDb.findById(inventory1.getSku())).thenReturn(Optional.of(product1));
+
+        // WHEN
+        productService.findProductAndAddInventory(inventory1);
+
+        // THEN
+        verify(productMongoDb).findById("123");
+        verify(productMongoDb).save(product1);
+        assertThat(product1.getInventory(), containsInAnyOrder(inventory1));
+    }
+
+
+    @Test
+    @DisplayName("if product is not available nothing should happen")
```

Comment:
available is confusing as the product may still exist in your DB but may only be not available for purchase
you may want to consider changing your description to exist in DB rather than available